### PR TITLE
AOCL: add v4.2.0

### PR DESCRIPTION
**Pull request by amd-toolchain-support**
This PR adds version 4.2 of the AMD AOCL Libraries.

* New library recipes: aocl-crypto, aocl-compression, aocl-libmem
* Updated recipes: CMake support in amdlibflame
* Improved vectorization behaviour to honor Spack target in "auto" mode
* Removed `verbose` variant in amdlibm which affects only build time behaviour

### CODE REVIEW

File: var/spack/repos/builtin/packages/amdlibflame/package.py [hunk 1]
```diff
@@ -94,27 +125,69 @@ def flag_handler(self, name, flags):
...
+class CMakeBuilder(spack.build_systems.cmake.CMakeBuilder):
+    def cmake_args(self):
+        spec = self.spec
+        args = [self.define("LIBAOCLUTILS_INCLUDE_PATH", self.spec["aocl-utils"].prefix.include)]
+        aocl_utils_lib_path = self.spec["aocl-utils"].libs
+        args.append("-DLIBAOCLUTILS_LIBRARY_PATH={0}".format(aocl_utils_lib_path))
+        # From 3.2 version, amd optimized flags are encapsulated under:
+        # ENABLE_AMD_AOCC_FLAGS for AOCC compiler
+        # ENABLE_AMD_FLAGS for all other compilers
+        if "@3.2:" in self.spec:
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/amdlibflame/package.py]: Version range-related checks are more intuitive when self.spec.satisfies()` is used (see docs and note at https://spack.readthedocs.io/en/latest/packaging_guide.html#testing-spec-constraints).

File: var/spack/repos/builtin/packages/aocl-compression/package.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/aocl-compression/package.py]: PR checks will let this pass at this time but `2024` as the ending year is preferred for new packages.

**Review by tldahlgren — DISMISSED** [review]
Confirmed the new version sha256 and homepages for new packages.  Have some minor suggestions related to use of `spec.satisfies()` and the ending copyright year for the new packages.

**Comment by tldahlgren** [comment]
Let me know if you will be addressing any of the suggestions or want the PR merged as-is.

**Comment by tldahlgren** [comment]
Or it can wait until @alalazo and or @adamjstewart comment since they've been explicitly requested to review (which I missed earlier).

**Comment by adamjstewart** [comment]
I don't know anything about any of these packages.

**Comment by amd-toolchain-support** [comment]
All the necessary changes have been made. so please merge this PR.

File: var/spack/repos/builtin/packages/aocl-compression/package.py (hunk 2 above)
Comment (amd-toolchain-support) [reply var/spack/repos/builtin/packages/aocl-compression/package.py]: Now the copyright year is changed to 2024.

File: var/spack/repos/builtin/packages/amdlibflame/package.py (hunk 1 above)
Comment (amd-toolchain-support) [reply var/spack/repos/builtin/packages/amdlibflame/package.py]: Suggested changes were incorporated.

File: var/spack/repos/builtin/packages/amdlibflame/package.py [hunk 3]
```diff
@@ -187,40 +187,39 @@ def configure_args(self):
         # From 3.2 version, amd optimized flags are encapsulated under:
         # enable-amd-aocc-flags for AOCC compiler
         # enable-amd-flags for all other compilers
-
-        if "@3.2:" in self.spec:
-            if "%aocc" in self.spec:
+        if spec.satisfies("@3.2: "):
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/amdlibflame/package.py:190]: The extra space is not required but I doubt it causes problems with the check.

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**